File: codeforces/houses/houses.py
n = int(input())
houses = [int(x) for x in input().split(' ')]

i = 0
j = n - 1
maximum = 0

if houses[i] == houses[j]:
    while houses[i] == houses[0] and j > i:
        i += 1
    while houses[j] == houses[0] and j > i:
        j -= 1
    maximum = i
    if (n - j - 1) < maximum:
        maximum = n - j - 1

print(n - 1 - maximum)

File: codeforces/houses/houses.js
var n = +(readline());
write(n);
write('\n');
var houses = readline().split(' ');

write(houses);
write('\n');
function search() {
    var step = 1;
    while (step <= n) {
        for (var i = 0; i < step; i++) {
            if (houses[i] != houses[i + (n - step)]) {
                return n - step
            }
        }
    }
    return 0;
}

write(search());
write('\n');